tests/ui/async-await/dont-ice-for-type-mismatch-in-closure-in-async.stderr

```diff
@@ -0,0 +1,46 @@
+error[E0308]: mismatched types
+  --> $DIR/dont-ice-for-type-mismatch-in-closure-in-async.rs:9:13
+   |
+LL | /         if true {
+LL | |             false
+   | |             ^^^^^ expected `()`, found `bool`
+LL | |
+LL | |         }
+   | |_________- expected this to be `()`
+
+error[E0308]: mismatched types
+  --> $DIR/dont-ice-for-type-mismatch-in-closure-in-async.rs:12:9
+   |
+LL |         true
+   |         ^^^^ expected `Option<()>`, found `bool`
+   |
+   = note: expected enum `Option<()>`
+              found type `bool`
+
+error[E0271]: expected `{closure@dont-ice-for-type-mismatch-in-closure-in-async.rs:6:10}` to be a closure that returns `bool`, but it returns `Option<()>`
```

Comment:
This error message is a bit unfortunate, but that can be addressed in the future.

Reply:
yeah; beta backports should be small, so I'm trying not to do a lot of cleanups here

Reply:
Yeah, makes sense, not really in the scope of this PR for fixing the ICE which is the bigger concern